deeptime/markov/msm/tram/_tram_model.py

```diff
@@ -0,0 +1,224 @@
+    """
+    def __init__(self, count_models, transition_matrices,
+                 biased_conf_energies,
+                 lagrangian_mult_log,
+                 modified_state_counts_log,
+                 therm_state_energies=None,
+                 markov_state_energies=None,
+                 ):
+        self.n_therm_states = biased_conf_energies.shape[0]
+        self.n_markov_states = biased_conf_energies.shape[1]
+
+        self._biased_conf_energies = biased_conf_energies
+        self._modified_state_counts_log = modified_state_counts_log
+        self._lagrangian_mult_log = lagrangian_mult_log
+        self._markov_state_energies = markov_state_energies
+
+        # if therm_state_energies is None:
+        self._therm_state_energies = np.asarray([-logsumexp(-row) for row in biased_conf_energies])
+        # else:
+        #     self._therm_state_energies = therm_state_energies
+
+        self._markov_state_model_collection = self._construct_markov_model_collection(
+            count_models, transition_matrices)
+
+    @property
+    def biased_conf_energies(self) -> np.ndarray:
+        r""" The estimated free energy per thermodynamic state and Markov state, :math:`f_k^i`, where :math:`k` is the
+        thermodynamic state index, and :math:`i` the Markov state index.
+        """
+        return self._biased_conf_energies
+
+    @property
+    def lagrangian_mult_log(self) -> np.ndarray:
+        r""" The estimated logarithm of the lagrange multipliers :math:`v_i^k` of each Markov state for all thermodynamic
+            states. lagrangian_mult_log[k,i] contains the lagrange multiplier of Markov state i in thermodynamic state k.
+        """
+        return self._lagrangian_mult_log
+
+    @property
+    def modified_state_counts_log (self):
+        r"""The logarithm of the modified state counts :math:`R_i^k` of each Markov state for all thermodynamic
+        states. modified_state_counts_log[k,i] contains the state counts of Markov state i in thermodynamic state k.
+        """
+        return self._modified_state_counts_log
+
+    @property
+    def therm_state_energies(self) -> np.ndarray:
+        r""" The estimated free energy per thermodynamic state, :math:`f_k`, where :math:`k` is the thermodynamic state
+        index.
+        """
+        return self._therm_state_energies
+
+
+    @property
+    def markov_state_energies(self) -> np.ndarray:
+        r""" The estimated free energy per Markov state, :math:`f^i`, where :math:`i` is the Markov state
+        index.
+        """
+        return self._markov_state_energies
+
+    @property
+    def markov_state_model_collection(self):
+        r""" The underlying MarkovStateModelCollection. Contains one Markov state model for each sampled thermodynamic
+        state.
+
+        Returns
+        -------
+        markov_state_model_collection : MarkovStateModelCollection
+            the collection of markov state models containing one model for each thermodynamic state.
```

Comment:
Documenting properties works a bit differently because of how sphinx handles things

```suggestion
        :getter: The collection of markov state models containing one model for each thermodynamic state.
        :type: MarkovStateModelCollection
```